src/modules/common/hooks/useDBConnection.ts

```diff
@@ -13,16 +13,23 @@ export const dataSource = new DataSource({
 })
 
 export const useDBConnection = () => {
+  const [isInitialized, setIsInitialized] = useState(false)
+  const [error, setError] = useState<Error | null>(null)
+
   useEffect(() => {
-    dataSource
-      .initialize()
-      .then(() => {
+    const initializeDB = async () => {
+      try {
+        logger.info('Initializing database...')
+        await dataSource.initialize()
         logger.info('DataSource has been initialized!')
-      })
-      .catch((err) => {
+        setIsInitialized(true)
+      } catch (err: any) {
         logger.error('Error during DataSource initialization:', err)
-      })
+        setError(err)
+      }
+    }
+    initializeDB()
```

Comment:
Doesn't DataSource have an error event that you can subscribe to, that you had to make a state for the flag and the error

Reply:
DataSource in TypeORM is not a React component or thread that supports events. Rather, it acts as a service to configure and manage the database connection.